packages/ra-core/src/form/Form.stories.tsx

```diff
@@ -403,3 +412,121 @@ export const ServerSideValidation = () => {
         </CoreAdminContext>
     );
 };
+
+export const MultiRoutesForm = ({
+    url,
+    initialRecord,
+}: {
+    url?: any;
+    initialRecord?: Partial<RaRecord>;
+}) => (
+    <TestMemoryRouter key={url} initialEntries={[url]}>
+        <CoreAdminContext i18nProvider={defaultI18nProvider}>
+            <Routes>
+                <Route
+                    path="/form/*"
+                    element={
+                        <RecordContextProvider value={initialRecord}>
+                            <FormWithSubRoutes />
+                        </RecordContextProvider>
+                    }
+                />
+            </Routes>
+        </CoreAdminContext>
+    </TestMemoryRouter>
+);
+
+MultiRoutesForm.args = {
+    url: 'unmodified',
+    initialRecord: 'none',
+};
+
+MultiRoutesForm.argTypes = {
+    url: {
+        options: [
+            'unmodified',
+            'modified with location state',
+            'modified with location search',
+        ],
+        mapping: {
+            unmodified: '/form/general',
+            'modified with location state': {
+                pathname: '/form/general',
+                state: { record: { body: 'from-state' } },
+            },
+            'modified with location search': `/form/general?source=${encodeURIComponent(JSON.stringify({ body: 'from-search' }))}`,
+        },
+        control: { type: 'select' },
+    },
+    initialRecord: {
+        options: ['none', 'provided'],
+        mapping: {
+            none: undefined,
+            provided: { title: 'lorem', body: 'unmodified' },
+        },
+        control: { type: 'select' },
+    },
+};
+
+const FormWithSubRoutes = () => {
+    return (
+        <Form>
+            <TabbedForm />
+            <SubmitButton />
```

Comment:
To mimic the true `<SaveButton>`, `<SubmitButton />` should use `dirtyFields` instead of `isDirty`, which may be different as we know.